pkg/ipam/allocator/podcidr/podcidr.go

```diff
@@ -849,6 +849,11 @@ func (n *NodesPodCIDRManager) allocateNext(nodeName string) (*nodeCIDRs, bool, e
 		nCIDRs.v6PodCIDRs = []*net.IPNet{v6CIDR}
 	}
 
+	if (nCIDRs == nil) {
+		err = fmt.Errorf("unable to allocate node cidr for the node %s. You may encounter a bug, please connect to the developer", nodeName)
```

Comment:
Adding an error here looks good and the first part of the message is fine. Regarding the second part, `You may encounter a bug, please connect to the developer`, this may not be very helpful. I suppose you mean users should file a bug report? It's fine to have reports, but it won't help users to debug their setup quickly.

Instead, it would be best to have a message telling what's wrong so users can fix it. Looking at the code at this location, it looks like we do not expect `nCIDRs` to be `nil` if no `err` has been returned. So there's probably a function where we can return `nil` for both the value and the error, and this is what have apparently happened in your case when you set an incorrect node mask size. It would be great to return a non-nil error at that stage.

Reply:
I my case, If return value `nCDIRs` and `err` is `ni`l, cilium will add a `nil nCIDRs` to the `.spec.Ipam.Pod CID Rs ` on 
 the `ciliumnodes.cilium.io`, just like this:
```
kubectl get ciliumnodes.cilium.io 
Name:         l-k8s-06.ops
Namespace:    
Labels:       beta.kubernetes.io/arch=amd64
              beta.kubernetes.io/os=linux
              kubernetes.io/arch=amd64
              kubernetes.io/hostname=l-k8s-06.sg-zl.ops
              kubernetes.io/os=linux
Annotations:  <none>
API Version:  cilium.io/v2
Kind:         CiliumNode
Metadata:
  Creation Timestamp:  2020-08-24T07:14:41Z
Spec:
  Addresses:
    Ip:    10.8.192.20
    Type:  InternalIP
  Azure:
  Encryption:
  Eni:
  Health:
  Ipam:
    Pod CID Rs:
      <nil> <----------------here
```

It will cause cilium-agent can't allocate ip address to the node, and struck and `Waiting for CRD ciliumnetworkpolicies.cilium.io to be available" subsys=cilium-operator-generic` forever.

In my case is I misconfig the subnodemasksize to **64**. But cluster size is **27**. The `ipam` didn't return any error, just allocate a `nil nCIDRs` without any errors. I also send a pull request here to fix this error https://github.com/cilium/ipam/pull/1, Maybe it has another bug on Ipam like this, So I think we‘d better add a error logs here. When it return both nil `nCIDRs` and nil `err`.   Help user to know what happend here.

How about print some infomation about ipam info? like this:
```IPAM Info: cluster_cidr=x.x.x.x/x, nodemask=xx, Please check your configuration is correct. If it is rgiht, you may encounter a bug, please connect to the developer```

Reply:
I had a look at your other PR on the `ipam` repo and it looks great, thanks for that! This is exactly what I meant. And I have no objection to adding another check here for safety.

Adding IMAP info is a good idea. I would probably just keep the first part of the message (`IPAM Info: cluster_cidr=x.x.x.x/x, nodemask=xx, Please check your configuration is correct.`).

Reply:
done